src/routers/films.js

```diff
@@ -0,0 +1,72 @@
+const express = require("express");
+const router = express.Router();
+const { films } = require("../../data");
+const {
+  checkForMissingFields,
+  checkForExistingEntry,
+  findById,
+  notFoundError,
+} = require("../../src/utils/utils");
+
+// GET
+
+router.get("/", (req, res) => {
+  // Gets all films or filtered by director if given in query params
+  if (req.query.director) {
+    const filteredFilmsByDirector = films.filter((film) =>
+      film.director.toLowerCase().includes(req.query.director.toLowerCase())
+    );
+    if (filteredFilmsByDirector.length) {
+      return res.json({ films: filteredFilmsByDirector });
+    }
+    throw new notFoundError();
+  }
```

Comment:
There's nothing wrong with this, I'm only commenting on this one so you can have a think about the situation.

This API spec only has the director filter - what if there were multiple possible filters? Would you continue to add if statements here? What if multiple filters could be combined, e.g. `director=someone&releaseYear=2010&genre=comedy`? How might you build a function that handles one and multiple filters?